packages/ariakit/src/dialog/dialog.tsx

```diff
@@ -380,36 +380,12 @@ export const useDialog = createHook<DialogOptions>(
     props = useWrapElement(
       props,
       (element) => {
-        const renderFocusTrap = (getTabbable: typeof getFirstTabbableIn) => {
-          if (state.visible && modal && !visibleModals.length) {
-            return (
-              <span
-                tabIndex={0}
-                style={{ position: "fixed", top: 0, left: 0 }}
-                aria-hidden
-                data-focus-trap=""
-                onFocus={() => {
-                  const dialog = ref.current;
-                  if (!dialog) return;
-                  const tabbable = getTabbable(dialog, true);
-                  if (tabbable) {
-                    tabbable.focus();
-                  } else {
-                    // Fallbacks to the dialog element.
-                    dialog.focus();
-                  }
```

Comment:
I hope I'm not missing this case 😅 When will this happen?

Reply:
This will happen when there's no tabbable element inside the dialog and the dialog itself is not tabbable (`tabIndex={-1}`).

Reply:
This is still a valid case (although we don't have tests for it yet).

Reply:
Updated the logic, hope that suffices the use case.